postgres_kernel/kernel.py

```diff
@@ -210,6 +210,8 @@ def do_execute(self, code, silent, store_history=True,
             if header is not None and len(rows) > 0:
                 self.send_response(self.iopub_socket, 'display_data', display_data(header, rows))
 
+            self._conn.commit()
+
```

Comment:
I believe this is better served by the existing autocommit flag. It defaults to true, but you can turn it on and off with
```sql
--autocommit: true
```

With your patch, users no longer have the choice.

Reply:
I noticed, but if it's default to true, then it seems not to work. I created a trigger in a cell and two tests in two cells below. The second test fails because the trigger is somehow no longer there (checked with pgAdmin UI).

Reply:
Hmm. I just tested and it seems to be working for me. I know that's not a satisfying response. Maybe you can give me a better idea of how to reproduce the issue?

![](https://trello-attachments.s3.amazonaws.com/58d428743111af1d0a20cf28/5c031c8885f8db0a7f03bfdc/2f79475aed09d156f3637fa151ed1f59/capture.png)

Reply:
![image](https://user-images.githubusercontent.com/19567683/49338947-f50a8700-f629-11e8-9645-c5ca9578cb09.png)
Here is the result I wanted.
But before, the second second test (third cell) didn't raise exception as expected.

You can use my whole notebook here for test: 
https://github.com/mhoangvslev/JupyterPostgreSQLBasics/blob/master/TP03.ipynb

Reply:
Okay, I released the patch before merging with your latest commit that fixes autocommit beheviour. I will drop the forced commit at the end :)